src/msg/simple/Pipe.cc

```diff
@@ -828,16 +828,16 @@ void Pipe::set_socket_options()
     int flag = 1;
     int r = ::setsockopt(sd, IPPROTO_TCP, TCP_NODELAY, (char*)&flag, sizeof(flag));
     if (r < 0) {
-      r = -errno;
-      ldout(msgr->cct,0) << "couldn't set TCP_NODELAY: " << cpp_strerror(r) << dendl;
+      ldout(msgr->cct,0) << "couldn't set TCP_NODELAY: "
```

Comment:
@tchaikov Oh, thanks for your review,  then maybe we should also save the `errno` at  https://github.com/ceph/ceph/pull/10297/files#diff-459b1eb43a0f5b6b6b430e3c67fee6ceL861  and  https://github.com/ceph/ceph/pull/10297/files#diff-459b1eb43a0f5b6b6b430e3c67fee6ceL873  in this function ?